tiledb/sm/c_api/tiledb.cc

```diff
@@ -4961,6 +4962,23 @@ int32_t tiledb_fragment_info_get_fragment_name(
   return TILEDB_OK;
 }
 
+int32_t tiledb_fragment_info_get_fragment_name_v2(
+    tiledb_ctx_t* ctx,
+    tiledb_fragment_info_t* fragment_info,
+    uint32_t fid,
+    tiledb_string_t** name) {
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, fragment_info) == TILEDB_ERR)
+    return TILEDB_ERR;
```

Comment:
There's no need to call `sanity_check(ctx)` any longer; it has become redundant. It hasn't been removed yet, mostly because it should be optimized out (although I haven't verified that) and there are a lot of occurrence of it. Under the remodeling rule (bring things up to present standard when modifying them),  the call should go.

There's no check for `name` against `nullptr`.